Providers/Scripts/2.4x-2.5x/Scripts/nxOMSGenerateInventoryMof.py

```diff
@@ -0,0 +1,217 @@
+    init_vars(Instances)
+    Set(FeatureName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+    return [0]
+
+def Test_Marshall(FeatureName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    init_vars(Instances)
+    return Test(FeatureName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+
+def Get_Marshall(FeatureName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    arg_names = list(locals().keys())
+    init_vars(Instances)
+    
+    CurrentInstances = Instances
+    FeatureName = protocol.MI_String(FeatureName)
+    Enable = protocol.MI_Boolean(Enable)
+    for instance in CurrentInstances:
+        instance['InstanceName'] = protocol.MI_String(instance['InstanceName'])
+        instance['ClassName'] = protocol.MI_String(instance['ClassName'])
+        if instance['Properties'] is not None and len(instance['Properties']):
+            instance['Properties'] = protocol.MI_StringA(instance['Properties'])
+    Instances = protocol.MI_InstanceA(CurrentInstances)
+    RunIntervalInSeconds = protocol.MI_Uint64(RunIntervalInSeconds)
+    Tag = protocol.MI_String(Tag)
+    Format = protocol.MI_String(Format)
+    FilterType = protocol.MI_String(FilterType)
+
+    if Configuration is None:
+        Configuration = []
+    if Configuration is not None and len(Configuration):
+        Configuration = protocol.MI_StringA(Configuration)
+
+    retd = {}
+    ld = locals()
+    for k in arg_names:
+        retd[k] = ld[k]
+    return 0, retd
+
+           
+def Set(FeatureName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration):
+    if Enable == True:
+         GenerateInventoyMOF(FeatureName, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+    else:
+         mof_file_path = inventoryMof_path + FeatureName + '.mof'
+         conf_file_path = inventoryMof_path + FeatureName + '.conf'
+         if os.path.isfile(mof_file_path):
+            os.remove(mof_file_path)
+
+         if os.path.isfile(conf_file_path):
+            os.remove(conf_file_path)
+    return [0]
+
+
+
+def Test(FeatureName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration):
+
+    shouldGenerateMofFile = TestGenerateInventoyMOF(FeatureName, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+
+    shouldGenerateConfFile = TestGenerateInventoryConf(FeatureName, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+
+    if(shouldGenerateMofFile or shouldGenerateConfFile):
+        return [-1]
+
+    return [0]
+
+
+
+def GenerateInventoyMOFContents(FeatureName, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration):
```

Comment:
The configuration can contain extra parameters that can used while generating the conf files. It is not used in the conf file for file change tracking, however varad pointed out that conf in the patch management would need some more info which can be passed on via Configuration. 

Since creating resource is a rather tedious process, I add this so we don't have to modify this resource just because conf/resource need more information.